### PR TITLE
Added support for FLUSHDB command

### PR DESCRIPTION
**Pull request by ashwaniYDV**
closes https://github.com/DiceDB/dice/issues/389
Redis FLUSHDB https://redis.io/docs/latest/commands/flushdb/

TODO: add unit tests and integration tests

### CODE REVIEW

**Comment by ashwaniYDV** [comment]
hi @JyotinderSingh do we have to handle ResetStore() differently for SYN and ASYNC ways for FLUSHDB command?

File: core/eval.go [hunk 1]
```diff
@@ -2287,3 +2287,26 @@ func evalLPOP(args []string, store *Store) []byte {
...
+	flushType := constants.Sync
+	if len(args) == 1 {
+		flushType = strings.ToUpper(args[0])
+	}
+
+	switch flushType {
+	case constants.Sync:
+		store.ResetStore()
+	case constants.Async:
+		store.ResetStore()
```
Comment (JyotinderSingh) [inline core/eval.go:2321]:
```suggestion
	case constants.Sync, constants.Async:
		store.ResetStore()
```

File: core/eval.go [hunk 2]
```diff
@@ -2287,3 +2287,26 @@ func evalLPOP(args []string, store *Store) []byte {
...
+	if len(args) > 1 {
+		return diceerrors.NewErrArity("FLUSHDB")
+	}
+
+	flushType := constants.Sync
+	if len(args) == 1 {
+		flushType = strings.ToUpper(args[0])
+	}
+
+	switch flushType {
```
Comment (JyotinderSingh) [inline core/eval.go:2317]:
```suggestion
        // TODO: Update this method to work with shared-nothing multithreaded implementation
	switch flushType {
```

**Comment by JyotinderSingh** [comment]
> hi @JyotinderSingh do we have to handle ResetStore() differently for SYN and ASYNC ways for FLUSHDB command?
> 
> From Redis doc: `Note: an asynchronous FLUSHALL command only deletes keys that were present at the time the command was invoked. Keys created during an asynchronous flush will be unaffected.`
> 
> From our implementation, it seems like both `FLUSHALL SYNC` and `FLUSHALL ASYNC` can be handles using store.ResetStore()

The implementation is fine for now, we will need to update it later when we start operating in multi-threaded mode.

**Review by JyotinderSingh — APPROVED** [review]
Thanks for adding support for this command @ashwaniYDV!

**Pull request merged**